test/e2e/csv_e2e_test.go

```diff
@@ -4204,29 +4204,78 @@ var _ = Describe("Disabling copied CSVs", func() {
 			},
 		},
 	}
+	BeforeEach(func() {
+		Eventually(func() error {
 
-	When("an operator is installed in AllNamespace mode", func() {
-		BeforeEach(func() {
-			Eventually(func() error {
-				if err := ctx.Ctx().Client().Create(context.TODO(), ns); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create ns: %v", err)
-					return err
-				}
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: genName("csv-toggle-test-"),
+				},
+			}
 
-				if err := ctx.Ctx().Client().Create(context.TODO(), &operatorGroup); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create og: %v", err)
-					return err
-				}
+			operatorGroup = operatorsv1.OperatorGroup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      genName("csv-toggle-test-"),
+					Namespace: ns.GetName(),
+				},
+			}
```

Comment:
Can we use the [`SetupGeneratedTestNamespace`](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/util.go#L954) helper function instead, which is a way to generate a test namespace + an OperatorGroup resource? That allows us to also use the [`TeardownNamespace`](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/util.go#L982) helper function to also collect test artifacts when an individual test case fails.

For a concrete example of this pattern: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/subscription_e2e_test.go#L58.

Reply:
Thanks!  I'll take a look.

Reply:
I had to tweak SetupGeneratedTestNamespace function a little bit for this test.

Reply:
Any reason you had to change around that helper function? It's not immediately clear to me when looking at these changes again, as the output of the previous and current implementation(s) all appear to have the same output:

- A namespace is created
- An OperatorGroup that targets the generated namespace is created

Am I missing something?